src/main/java/updater/UpdateManager.java

```diff
@@ -97,12 +131,15 @@ private boolean initUpdate() {
             return false;
         }
 
+        updateConfig.setLastAppUpdateDownloadSuccessful(false);
+        saveUpdateConfig();
+
         return extractJarUpdater();
     }
 
     private boolean extractJarUpdater() {
```

Comment:
See if you can exceptions instead of booleans to indicate exceptional situations.